iothub/device/samples/how to guides/CustomSerializer/ProtobufPayloadConvention.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Google.Protobuf;
+using Microsoft.Azure.Devices.Client;
+
+namespace CustomSerializer
+{
+    internal class ProtobufPayloadConvention : PayloadConvention
+    {
+        private readonly MessageParser _parser;
+        public ProtobufPayloadConvention(MessageParser parser) => _parser = parser;
+        
+        public override string ContentType => "proto";
+
+        public override Encoding ContentEncoding => Encoding.Default;
+
+        public override T GetObject<T>(string jsonObjectAsText)
+        {
+            throw new NotImplementedException();
```

Comment:
This one is awkward. Let me see why I kept it in and see if I can get rid of it.

Reply:
I realize why. It is used after we've pulled out the twin properties (we have text) and need to deserialize to an object.

Since Twin only works with json, this should be good. I'd expect a protobuff implementor to decide how they want to deserialize their twin properties.